### PR TITLE
Fix bug in filter-clues plugin

### LINKED ISSUE #1758: Error in `filter-clues` Plugin
### Description

Running the following Command:

`./scancode -clp --json-pp sample_filter_clues.json samples --filter-clues`

Gives the Error:

```
ERROR: failed to run post-scan plugin: filter-clues:
Traceback (most recent call last):
  File "/home/ayansm/Desktop/GSoD/scancode-toolkit-versions/scancode-toolkit-3.1.1/src/scancode/cli.py", line 1033, in run_codebase_plugins
    plugin.process_codebase(codebase, **kwargs)
  File "/home/ayansm/Desktop/GSoD/scancode-toolkit-versions/scancode-toolkit-3.1.1/src/cluecode/plugin_filter_clues.py", line 90, in process_codebase
    filtered = filter_ignorable_resource_clues(resource, rules_by_id)
  File "/home/ayansm/Desktop/GSoD/scancode-toolkit-versions/scancode-toolkit-3.1.1/src/cluecode/plugin_filter_clues.py", line 105, in filter_ignorable_resource_clues
    resource.emails = filtered.emails
AttributeError: 'ScannedResource' object has no attribute 'emails'

Scanning done.
Some files failed to scan properly:
ERROR: failed to run post-scan plugin: filter-clues:
Traceback (most recent call last):
  File "/home/ayansm/Desktop/GSoD/scancode-toolkit-versions/scancode-toolkit-3.1.1/src/scancode/cli.py", line 1033, in run_codebase_plugins
    plugin.process_codebase(codebase, **kwargs)
  File "/home/ayansm/Desktop/GSoD/scancode-toolkit-versions/scancode-toolkit-3.1.1/src/cluecode/plugin_filter_clues.py", line 90, in process_codebase
    filtered = filter_ignorable_resource_clues(resource, rules_by_id)
  File "/home/ayansm/Desktop/GSoD/scancode-toolkit-versions/scancode-toolkit-3.1.1/src/cluecode/plugin_filter_clues.py", line 105, in filter_ignorable_resource_clues
    resource.emails = filtered.emails
AttributeError: 'ScannedResource' object has no attribute 'emails'
```

### System configuration
For bug reports, it really helps us to know:
* What OS are you running on? Ubuntu 16.04 LTS
* What version of scancode-toolkit was used to generate the scan file? v3.1.1
* What installation method was used to install/run scancode? Source Download

### PR DESCRIPTION
**Pull request by tushar912**
Fixes #1758 

### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁

### CODE REVIEW

File: tests/scancode/test_cli.py [hunk 1]
```diff
@@ -109,7 +109,7 @@ def test_verbose_option_with_copyrights(monkeypatch):
     assert len(open(result_file).read()) > 10
 
 
-@pytest.mark.xfail(reason='Bug is not fixed yet')
+
```
Comment (AyanSinhaMahapatra) [inline tests/scancode/test_cli.py]: A nitpick, keep only one new line between methods and two new lines between functions and classes.

**Comment by AyanSinhaMahapatra** [comment]
@tushar912 Does `./scancode -clpieu --json-pp sample_filter_clues.json samples --filter-clues` run properly now? Could you post the output JSON here?

The 2 failing tests seem to be from other problems that are fixed in the release branch.

**Comment by AyanSinhaMahapatra** [comment]
Also, could you change the commit message according to this doc - https://aboutcode.readthedocs.io/en/latest/contributing/writing_good_commit_messages.html

So, 
1. Add more context about what is changed in the commit body, 
2. change `fixes` to `fix` in both PR name and commit header. https://chris.beams.io/posts/git-commit/#imperative
etc.

**Comment by tushar912** [comment]
@AyanSinhaMahapatra I think it is working I am getting this json output

[filter_clues_bugfix_sample.json.txt](https://github.com/nexB/scancode-toolkit/files/5814042/filter_clues_bugfix_sample.json.txt)

**Comment by pombredanne** [comment]
@tushar912 @AyanSinhaMahapatra what's left to do here? I would like to merge this ASAP :)

**Review by AyanSinhaMahapatra — CHANGES_REQUESTED** [review]
@pombredanne I've tested this branch locally, the bug isn't there anymore, the test reproducing the earlier bug passes (the test fails are from other problems fixed in the release branch), and the plugin seems to be doing everything as it should, i.e. removing redundant duplicated clues already contained in detected license and copyright texts and notices. It seems ready to merge to me, but you should still check the changes to see everything is in place.

But the commit message has to be changed as it seems to have leftovers from squashing two commits together, having two bodies, and two signoffs. @tushar912

**Comment by tushar912** [comment]
@AyanSinhaMahapatra I have changed the commit message .I think it looks better now.

**Comment by AyanSinhaMahapatra** [comment]
Another nitpick, adding the issue number on the commit header/body would be best. Just add `#1758` after bug in the header and that would be all. Sorry I missed this before. Thanks++

**Comment by tushar912** [comment]
@AyanSinhaMahapatra I have made the change.

**Review by AyanSinhaMahapatra — APPROVED** [review]

**Review by pombredanne — APPROVED** [review]
LGTM! Thank you++

**Comment by pombredanne** [comment]
@AyanSinhaMahapatra please go ahead and merge

**Comment by pombredanne** [comment]
@tushar912 Thank you ++ :bow:

**Pull request merged**